packages/common/src/signHelpers.ts

```diff
@@ -3,40 +3,175 @@ import {
+        : value.accountKeys;
+
+    return Object.entries(keys).reduce(
+        (acc, [k, v]) => ({
+            ...acc,
+            [k]: Object.entries(v.keys).reduce(
+                (a, e) => ({ ...a, [e[0]]: e[1].signKey }),
+                {}
+            ),
+        }),
+        {}
+    );
+};
+
+const getCredentialSignature = async (
+    digest: Buffer,
+    keys: Record<number, HexString>
+): Promise<CredentialSignature> => {
+    const sig: CredentialSignature = {};
+    for (const key in keys) {
+        sig[key] = await getSignature(digest, keys[key]);
+    }
+    return sig;
+};
+
+/**
+ * Creates an `AccountSigner` for an account exported from a Concordium wallet.
+ * Creating signatures using the `AccountSigner` will hold signatures for all credentials and all their respective keys included in the export.
+ *
+ * @param {WalletExportFormat} walletExport - The wallet export object.
+ *
+ * @returns {AccountSigner} An `AccountSigner` which creates signatures using all keys for all credentials
+ */
+export function buildAccountSigner(
+    walletExport: WalletExportFormat
+): AccountSigner;
+/**
+ * Creates an `AccountSigner` for an arbitrary format extending the {@link WithAccountKeys} type.
+ * Creating signatures using the `AccountSigner` will hold signatures for all credentials and all their respective keys included.
+ *
+ * @param {AccountKeys} value.accountKeys - Account keys of type {@link AccountKeys} to use for creating signatures
+ *
+ * @returns {AccountSigner} An `AccountSigner` which creates signatures using all keys for all credentials
+ */
+export function buildAccountSigner<T extends WithAccountKeys>(
+    value: T
+): AccountSigner;
+/**
+ * Creates an `AccountSigner` for the {@link SimpleAccountKeys} type.
+ * Creating signatures using the `AccountSigner` will hold signatures for all credentials and all their respective keys included.
+ *
+ * @param {SimpleAccountKeys} keys - Account keys to use for creating signatures
+ *
+ * @returns {AccountSigner} An `AccountSigner` which creates signatures using all keys for all credentials
+ */
+export function buildAccountSigner(keys: SimpleAccountKeys): AccountSigner;
+/**
+ * Creates an `AccountSigner` for an account which uses the first credential's first keypair.
+ * Note that if the account has a threshold > 1 or the first credentials has a threshold > 1, the transaction signed using this will fail.
+ *
+ * @param {HexString} key - The ed25519 private key in HEX format. (First credential's first keypair's private key)
+ *
+ * @returns {AccountSigner} An `AccountSigner` which creates a signature using the first credentials first keypair
+ */
+export function buildAccountSigner(key: HexString): AccountSigner;
+export function buildAccountSigner<T extends WithAccountKeys>(
+    value: T | WalletExportFormat | SimpleAccountKeys | string
+): AccountSigner {
+    if (typeof value === 'string') {
+        return buildBasicAccountSigner(value);
+    }
```

Comment:
I'm not sure I understand this monolith function needs to handle the "basic format"? :thinking:

Reply:
Well, I thought it'd make more sense for the user to have an overloaded function that could do it all instead? In my head, it's easier to have different signatures for the same function instead of going looking for a different function.

Also adapted the readme to use the same function name for both single and multi-key signers